chore: initial event hook and Slack API spec

blefebvre opened the pull request:
Spec two new endpoints to support the automated creation of project collaboration channels (Slack), and the retrieval of their metadata for a particular IMS org.

Open question on the `GET /slack/{imsOrgId}` endpoint:

- Should we distinguish between `404`s where a SpaceCat Organization cannot be found for the given IMS org, vs. the case where a SpaceCat Organization was found but no Slack config was associated with it?

On the `POST /event/fulfillment` endpoint:

- Typically, only 1 event will be received at a time, but it is possible that there is more than 1 event in the fulfillment_completed payload.
- If there is 1 event in the payload which is NOT accepted for processing, and we do not respond to Hoolihan with a `2xx` response, the entire event payload will be sent again (see Hoolihan "Working with the Push Stream (Subscriber) API" doc)
- Knowing this, our best bet may be to log the failing event payload and still return a `202` response — but I am open to your input on this.

solaris007 commented:
> Should we distinguish between 404s where a SpaceCat Organization cannot be found for the given IMS org, vs. the case where a SpaceCat Organization was found but no Slack config was associated with it?

I would recommend aligning with the resource-centric nature of the API:
- The slack configuration is part of the organization resource (and for later: the site resource)
- The `imsOrgId` of an organization is optional and is therefore de facto not considered a primary key
- The organization's config could be considered a nested resource
- If present, the slack configuration resides within the `config` attribute of the organization's config resource
- We may or may not provide a convenience "shortcut" to the slack configuration

**The standard way**:
So far we have usually denoted lookup by an "alternate primary key" like this:
- `GET /organizations/by-ims-org-id/{imsOrgId}` (return 404 if not found)
- A client may then look at the `config` attribute and check the config's `slack` property

**The convenient way**:
This one would conveniently absolve the client from understanding an organization's config schema and return slack information directly.
- `GET /organizations/by-ims-org-id/{imsOrgId}/slackConfig` (return 404 if *organization* not found)
- If no config is set or no slack config, this could return a `204` no content status,  which explicitly indicates the absence of content - in this case the (slack-) config.

Thoughts?

solaris007 commented:
> On the POST /event/fulfillment endpoint:
> 
> Typically, only 1 event will be received at a time, but it is possible that there is more than 1 event in the fulfillment_completed payload.
> If there is 1 event in the payload which is NOT accepted for processing, and we do not respond to Hoolihan with a 2xx response, the entire event payload will be sent again (see Hoolihan "Working with the Push Stream (Subscriber) API" doc)
> Knowing this, our best bet may be to log the failing event payload and still return a 202 response — but I am open to your input on this.

I completely agree that we should absolve Hoolihan from any further responsibility or retry and handle this internally. Logging seems to fine, especially in the context of the powers of Coralogix we have to enable log pattern detection and alerting/graphing.

solaris007 commented on docs/openapi/schemas.yaml:
```diff
@@ -438,3 +437,167 @@ AuditList:
+      - status
+    properties:
+      status:
+        description: The processing status (whether or not the event was accepted for processing)
```
I would not call it "processing status" but rather something like "acceptance status" (since we're really not returning any status of the subsequent processing)

solaris007 reviewed (commented):
@blefebvre great contribution! For me the only open point is the discussion in https://github.com/adobe/spacecat-api-service/pull/121#issuecomment-1911447974 - otherwise LGTM

blefebvre commented:
Thanks @solaris007 for the review and discussion.

- Replaced `/slack/{imsOrgId}` with `/organizations/by-ims-org-id/{imsOrgId}` 
- Added convenience endpoint `/organizations/by-ims-org-id/{imsOrgId}/slack-config` to the spec

I like how this is coming together: 
<img width="383" alt="image" src="https://github.com/adobe/spacecat-api-service/assets/1048207/7921793a-054f-4736-a38b-821da15d2cbd">

solaris007 reviewed (approved):
Looking great - thanks for your contribution :)

solaris007 commented:
:tada: This PR is included in version 1.17.3 :tada:

The release is available on:
- `v1.17.3`
- [GitHub release](https://github.com/adobe/spacecat-api-service/releases/tag/v1.17.3)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
